lib/utils/token_utilities.dart

```diff
@@ -216,10 +217,76 @@ class TokenUtilities {
     return 1;
   }
 
-  /// Checks if two currencies match (by title and tag)
-  static bool _matchesToken(Erc20Token token, CryptoCurrency currency) {
-    return token.title.toLowerCase() == currency.title.toLowerCase() &&
-        (token.tag?.toLowerCase() == currency.tag?.toLowerCase() ||
-            (token.tag == null && currency.tag == null));
+  static Future<List<CryptoCurrency>> getAvailableTokensForNetwork(
+    WalletType network,
+  ) async {
+    final baseCurrency = walletTypeToCryptoCurrency(network);
+    final allTokens = <CryptoCurrency>[];
+    final addedAddresses = <String>{};
+
+    allTokens.add(baseCurrency);
+
+    for (final currency in CryptoCurrency.all) {
+      // For EVM networks: ETH has no tag, POL/BASE have tags
+      // Match by tag for POL/BASE, match by title==tag for ETH
+      final matches = (baseCurrency.tag == null && baseCurrency.title == currency.tag) ||
+          (baseCurrency.tag != null &&
+              currency.tag?.toLowerCase() == baseCurrency.tag?.toLowerCase());
+
+      if (matches && _shouldAddToken(allTokens, currency, addedAddresses)) {
+        allTokens.add(currency);
+        if (currency is Erc20Token) {
+          addedAddresses.add(currency.contractAddress.toLowerCase());
+        }
```

Comment:
currency will never be `ERC20Token` you are looping on `CryptoCurrency.all` which doesn't have any ERC20Token, it would still work since addedAddresses will always be empty, but just stating that this is useless

Reply:
Yeah that makes sense

We don't need the check there. I'll take it out